### PR TITLE
feat: 🎸 migrate blog settings from remark to contentlayer

### PR DESCRIPTION
**Pull request by abrahamcalsin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,3 +19,33 @@ git clone https://github.com/abrahamcalsin/abrahamcalsin.com
...
+
+1. https://techinplanet.com/cant-match-page-to-uri-with-next-js-error-the-provided-path-does-not-match-the-page-slug/
+2. https://github.com/vercel/next.js/discussions/16881
+3. https://nextjs.org/docs/basic-features/data-fetching/get-static-props
+4. https://nextjs.org/docs/messages/incompatible-href-as
+5. https://imadatyatalah.vercel.app/guides/how-to-setup-contentlayer-in-nextjs
+6. https://dev.to/imadatyat/how-to-setup-contentlayer-in-your-nextjs-app-3ilp
+7. https://github.dev/rsipakov/next-pwa-contentlayer
+8. https://devdojo.com/jeblister/building-a-podcast-site-with-nextjs-mdx-and-contentlayer
+9. https://blog.mrcatdev.com/turns-your-mdx-files-include-yaml-into-data-with-contentlayer-in-nextjs-100daysofcode
```
Comment (yonycalsin) [inline README.md:51]: I think, these resources should be upload in gists, `https://gist.github.com/`

File: contentlayer.config.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { defineDocumentType, makeSource } from 'contentlayer/source-files'
+import readingTime from 'reading-time'
+
+export const Blog = defineDocumentType(() => ({
+  name: 'Blog',
+  // filePathPattern: `blog/*.md`,
```
Comment (yonycalsin) [inline contentlayer.config.ts:6]: remove this commented line
```suggestion
```

File: next.config.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
+const { withContentlayer } = require('next-contentlayer')
+
 /** @type {import('next').NextConfig} */
-module.exports = {
+module.exports = withContentlayer()({
   reactStrictMode: true,
-  swcMinify: true,
-}
+  // swcMinify: true,
```
Comment (yonycalsin) [inline next.config.js:6]: remove this commented line

```suggestion
```

File: pages/blog/[slug].tsx [hunk 4]
```diff
@@ -28,61 +29,73 @@ const Article = (props: ArticleProps) => {
...
+): Promise<GetStaticPropsResult<ArticleProps>> {
+  const { params } = context
+
+  const article = allBlogs.find(article => {
+    const [, onlySlug] = article.slug.split('/')
+
+    return onlySlug === params?.slug
+  })
+
+  // console.log(123, { params, hasArticle: !!article })
```
Comment (yonycalsin) [inline pages/blog/[slug].tsx:88]: remove this commented line

```suggestion
```

**Review by yonycalsin — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (abrahamcalsin) [reply README.md:51]: > I think, these notes should be upload to github-gists like a secret-gist, `https://gist.github.com/`

Okay, thank you, I have just modified it. 🥳

File: contentlayer.config.ts (hunk 2 above)
Comment (abrahamcalsin) [reply contentlayer.config.ts:6]: well, thanks

File: next.config.js (hunk 3 above)
Comment (abrahamcalsin) [reply next.config.js:6]: well, thanks

**Review by yonycalsin — APPROVED** [review]
Good Job 👍

**Pull request merged**